FindClang.cmake

```diff
@@ -82,7 +82,7 @@ endif()
 
 if(NOT Clang_EXECUTABLE STREQUAL "Clang_EXECUTABLE-NOTFOUND")
   execute_process(
-    COMMAND clang --version
+    COMMAND ${Clang_EXECUTABLE} --version
```

Comment:
you call find_program(Clang_EXECUTABLE
  clang ... earlier, so what difference does it make if you use clang / Clang_EXECUTABLE here.. am i missing something?

Reply:
`find_program` searches for the program and stores it's value as the first variable passed to `find_program`
So in the case of, 
```
find_program(Clang_EXECUTABLE
  clang
  PATHS ${CMAKE_SOURCE_DIR}/stm32-tools/clang/${CMAKE_HOST_SYSTEM_NAME}
  PATH_SUFFIXES bin)
```
We are searching for clang inside `ot3-firmware/stm32-tools/clang/Linux` and storing it to the `Clang_EXECUTABLE` variable. 

I am assuming that on Mac the clang executable is in fact just `clang` whereas on my Linux system it is `ot3-firmware/stm32-tools/clang/Linux/clang`